dill/_dill.py

```diff
@@ -447,6 +446,44 @@ def load_session(filename='/tmp/session.pkl', main=None, **kwds):
 
 ### End: Pickle the Interpreter
 
+class sentinel(object):
+    """
+    Create a unique sentinel object that is pickled as a constant.
+    """
+    def __init__(self, name):
+        self.name = name
+    def __repr__(self):
+        return __name__ + '.' + self.name # pragma: no cover
+    def __copy__(self):
+        return self # pragma: no cover
+    def __deepcopy__(self, memo):
+        return self # pragma: no cover
+    def __reduce__(self):
+        return self.name
+    def __reduce_ex__(self, protocol):
+        return self.name
+
+class BuiltinShim(object):
```

Comment:
Probably could just be called `Shim` or `FunctionShim`.  There's nothing that requires a `builtin`, actually.  It's more that if a named shim exists (i.e is in the chosen shim dict), then use it... otherwise use the provided alternate function, whatever that may be.